modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/security/OAuthConfig.java

```diff
@@ -0,0 +1,89 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.crypto.factory.PasswordEncoderFactories;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerSecurityConfigurer;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.oauth2.provider.token.store.JwtAccessTokenConverter;
+import org.springframework.security.oauth2.provider.token.store.JwtTokenStore;
+import com.expedia.adaptivealerting.modelservice.service.UserDetailsServiceImpl;
+
+/**
+ * Configure the authorization server.
+ *
+ * @author kashah
+ */
+@Configuration
+@EnableAuthorizationServer
+public class OAuthConfig extends AuthorizationServerConfigurerAdapter {
+
+    @Autowired
+    @Qualifier("authenticationManagerBean")
+    private AuthenticationManager authenticationManager;
+
+    @Autowired
+    private UserDetailsServiceImpl userDetailsService;
+
+    @Autowired
+    private SecurityProperties properties;
+
+    @Override
+    public void configure(final AuthorizationServerSecurityConfigurer oauthServer) throws Exception {
+        oauthServer.tokenKeyAccess("permitAll()").checkTokenAccess("isAuthenticated()");
+    }
+
+    @Override
+    public void configure(ClientDetailsServiceConfigurer clients) throws Exception {
+        clients.inMemory()
+                .withClient(properties.getClientId()).secret((passwordEncoder().encode(properties.getSecret())))
+                .authorizedGrantTypes("password", "authorization_code", "refresh_token").scopes("read", "write")
+                .autoApprove(true);
+    }
+
+    @Override
+    public void configure(final AuthorizationServerEndpointsConfigurer endpoints) throws Exception {
+        endpoints.tokenStore(tokenStore()).authenticationManager(authenticationManager).accessTokenConverter(defaultAccessTokenConverter())
+                .userDetailsService(userDetailsService);
+    }
+
+    @Bean
+    public TokenStore tokenStore() {
+        return new JwtTokenStore(defaultAccessTokenConverter());
+    }
+
+    @Bean
+    public JwtAccessTokenConverter defaultAccessTokenConverter() {
+        JwtAccessTokenConverter converter = new JwtAccessTokenConverter();
+        converter.setSigningKey("123");
```

Comment:
Need an actual secret key here, not "123". I assume you just copied this from the Baeldung tutorial (https://www.baeldung.com/spring-security-oauth-jwt).

See https://docs.spring.io/spring-security/oauth/apidocs/org/springframework/security/oauth2/provider/token/store/JwtAccessTokenConverter.html#setSigningKey(java.lang.String) for more information